extensions/amp-gwd-animation/0.1/amp-gwd-animation.js

```diff
@@ -110,13 +118,23 @@ export class GwdAnimation extends AMP.BaseElement {
    */
   createAction_(actionName) {
     return invocation => {
+      // The setCurrentPage action is special-cased, because the event which
+      // triggers it (slideChange) may be emitted by an amp-carousel other
+      // than the GWD pagedeck. Ignore the event in this case.
+      if (actionName == 'setCurrentPage') {
+        const gwdPageDeck = this.getGwdPageDeck_();
+        if (!(gwdPageDeck && invocation.source == gwdPageDeck)) {
+          return;
+        }
```

Comment:
Nit: If you hoist to another method maybe you could use a shorter comment or even omit it.

```javascript
  return invocation => {
    if (this.shouldIgnoreInvocation_(actionName, invocation)) {
       return;
    }
    // ..
  };
}
// ...
shouldIgnoreInvocation_(actionName, invocation) {
  const pageDeck = this.getGwdPageDeck_();
  const comesFromPageDeck = pageDeck && invocation.source == pageDeck;
  return actionName == 'setCurrentPage' && !comesFromPageDeck;
}
```

Reply:
I split up the action handler into separate shouldExecuteInvocation and executeInvocation steps, which I think makes it more expressive. actionHandler_ is now used as a generic action handler rather than returning a handler function bound to the action name, since the action name is retrievable from the invocation detail.